vitrina/datasets/views.py

```diff
@@ -1205,14 +1206,28 @@ def get_history_objects(self):
         plan_history_objects = Version.objects.get_for_model(Plan).filter(
             object_id__in=list(dataset_plan_ids)
         )
+        dataset_distribution_history_objects = self._get_history_objects_for_model(DatasetDistribution)
+        attribution_history_objects = self._get_history_objects_for_model(DatasetAttribution)
+        relation_history_objects = self._get_history_objects_for_model(DatasetRelation)
 
         history_objects = (
             property_history_objects
             | model_history_objects
             | dataset_history_objects
             | plan_history_objects
+            | dataset_distribution_history_objects
+            | attribution_history_objects
+            | relation_history_objects
         )
         return history_objects.order_by("-revision__date_created")
+    
+    def _get_history_objects_for_model(self, model: TypingType[models.Model]) -> list[int]:
+        version_ids = [
+            version.pk for version in Version.objects.get_for_model(model)
+            if version.field_dict['dataset_id'] == self.object.id
+        ]
+
+        return Version.objects.get_for_model(model).filter(pk__in=version_ids)
```

Comment:
This seems redundant, I don't see a reason to call `Version.objects.get_for_model(model)` twice. If a list of versions would suffice the needs then:

```suggestion
    return [
        version for version in Version.objects.get_for_model(model)
        if version.field_dict.get('dataset_id') == self.object.id
    ]
```

If not it could at least be shortened (I think it would still be two calls here):

```suggestion
    versions = Version.objects.get_for_model(model)
    filtered_ids = [
        version.pk for version in versions
        if version.field_dict.get('dataset_id') == self.object.id
    ]
    return versions.filter(pk__in=filtered_ids)
```

Reply:
I need `QuerySet` for return, thats why first option doesn't work. And I don't understand how second option is shorter? :confused: